packages/core/lib/patchers/aws3_p.ts

```diff
@@ -62,110 +79,115 @@ function addFlags(http: HttpResponse, subsegment: Subsegment, err?: SdkError): v
-      }
-      return send.apply(client, args) as Promise<Output>;
-    }
-
-    const subsegment = segment.addNewSubsegment(serviceIdentifier);
-    subsegment.addAttribute('namespace', 'aws');
-
-    try {
-      const res = (await send.apply(client, [command])) as Output;
-
-      if (!res) throw new Error('Failed to get response from instrumented AWS Client.');
-
-      const [aws, http] = await buildAttributesFromMetadata(
-        client,
-        command,
-        res.$metadata,
-      );
-
-      subsegment.addAttribute('aws', aws);
-      subsegment.addAttribute('http', http);
-
-      addFlags(http, subsegment);
-      subsegment.close();
-      return res;
-    } catch (err) {
-      if (err.$metadata) {
-        const [aws, http] = await buildAttributesFromMetadata(
-          client,
-          command,
-          err.$metadata,
-        );
-
-        subsegment.addAttribute('aws', aws);
-        subsegment.addAttribute('http', http);
-        addFlags(http, subsegment, err);
-      }
-
-      const errObj = { message: err.message, name: err.name, stack: err.stack || new Error().stack };
-
-      subsegment.close(errObj, true);
-      throw err;
-    }
-  };
-
-  client.middlewareStack.add(
-    (next) => async (args: any) => {
-      const segment = manualSegment || contextUtils.resolveSegment();
-      if (!segment) return next(args);
-
-      const parent = (segment instanceof Subsegment
-        ? segment.segment
-        : segment);
-
-      args.request.headers['X-Amzn-Trace-Id'] = stringify(
+const getXRayPlugin = (config: RegionResolvedConfig, manualSegment?: SegmentLike): Pluggable<any, any> => {
+  const XRayPluginImpl: Pluggable<any, any> = {
+    applyToStack: (stack) => {
+      stack.add(
+        (next: any, context: any) => async (args: any) => {
+          const segment = contextUtils.isAutomaticMode() ? contextUtils.resolveSegment() : manualSegment;
+          const {clientName, commandName} = context;
+          const operation = commandName.slice(0, -7); // Strip trailing "command" string
+          const service = clientName.slice(0, -6);    // Strip trailing "client" string
+  
+          if (!segment) {
+            const output = service + '.' + operation;
+      
+            if (!contextUtils.isAutomaticMode()) {
+              logger.getLogger().info('Call ' + output + ' requires a segment object' +
+                ' on the request params as "XRaySegment" for tracing in manual mode. Ignoring.');
+            } else {
+              logger.getLogger().info('Call ' + output +
+                ' is missing the sub/segment context for automatic mode. Ignoring.');
+            }
+            return next(args);
+          }
+  
+          const subsegment: Subsegment = segment.addNewSubsegment(service);
+          subsegment.addAttribute('namespace', 'aws');
+  
+          const parent = (segment instanceof Subsegment
+            ? segment.segment
+            : segment);
```

Comment:
AWS SDK for JavaScript team recommends using [prettier](https://prettier.io/) for formatting. 
```suggestion
          const parent = segment instanceof Subsegment ? segment.segment : segment;
```

Your package.json already has prettier, but it wasn't run on this file.
https://github.com/aws/aws-xray-sdk-node/blob/8fa31d45c6276912f36de779f376c7cddb08446e/package.json#L39

I recommend adding a [pre-commit](https://prettier.io/docs/en/precommit.html) hook to run prettier. It can be done as a different PR.